### PR TITLE
sdk/state: add helper methods for building xdr strings in tests

### PR DESCRIPTION
**Pull request by acharb**
**WHAT**
adds helper methods to be used in tests for creating desired transaction result xdr and transaction meta xdr.

**WHY**
the current xdr strings used in the tests work but are difficult to adjust when data needs to be changed

closes https://github.com/stellar/experimental-payment-channels/issues/269

note: I didn't change all the current xdr strings in the tests to use these new helper methods. I changed one to show how the helper methods could be used, and the rest can be changed on a need basis.

### CODE REVIEW

File: sdk/state/ingest_test.go
Comment (leighmcculloch) [inline sdk/state/ingest_test.go:860]: 💡 I think we might be able to get away with only supporting `...interface{}`, and the function can assume it's the after state we're wanting to set and pass the value into `xdr.NewLedgerEntryData`. It would reduce the test code down to:
```suggestion
	resultMetaXDR, err := txbuildtest.BuildResultMetaXDR(
		&xdr.AccountEntry{
			AccountId: xdr.MustAddress(initiatorEscrow.Address()),
			SeqNum:    102,
			Signers: []xdr.Signer{
				{ Key: xdr.MustSigner("GAKDNXUGEIRGESAXOPUHU4GOWLVYGQFJVHQOGFXKBXDGZ7AKMPPSDDPV"), Weight: 1 },
				},
				Thresholds: xdr.Thresholds{0, 2, 2, 2},
			},
		},
		&xdr.AccountEntry{
			AccountId: xdr.MustAddress(responderEscrow.Address()),
		},
	})
```

File: sdk/state/ingest_test.go
Comment (leighmcculloch) [inline sdk/state/ingest_test.go:840]: 💡 There are some helpers for some XDR types in the xdr package, so you can shorten this a little.
```suggestion
								Key: xdr.MustSigner("GAKDNXUGEIRGESAXOPUHU4GOWLVYGQFJVHQOGFXKBXDGZ7AKMPPSDDPV"),
```

**Review by leighmcculloch — APPROVED** [review]
Looks great. I left a couple suggestions (💡) that I think might reduce the boilerplate that needs repeating in tests, but defer to you.

File: sdk/txbuildtest/txbuildtest.go
Comment (acharb) [reply sdk/txbuildtest/txbuildtest.go]: https://github.com/stellar/experimental-payment-channels/pull/271/commits/a9e88491513cf0f195a9b6a9d63600cd3d112694

File: sdk/state/ingest_test.go
Comment (acharb) [reply sdk/state/ingest_test.go]: https://github.com/stellar/experimental-payment-channels/pull/271/commits/55d2381e67f9855f83edc2a4bb7d529262f52553

File: sdk/state/ingest_test.go
Comment (acharb) [reply sdk/state/ingest_test.go]: I couldn't think of a good way of fitting `xdr. NewLedgerEntryData` into it, since we may need to pass in multiple ledgerEntryData object of different types (like in `TestChannel_IngestTx_updateState_invalid_initiatorEscrowHasExtraSigner`)

but I changed it to assume it's an update change so we only have to pass in `[]xdr.LedgerEntryData` now, which also really simplifies what we have to pass in.

going to merge since approved but if you see something that can be improved lmk

Comment (leighmcculloch) [reply sdk/state/ingest_test.go]: If you change the input type to `entries ...interface{}`, and then inside the function: 
```go
for _, e := range entries {
	switch le := e.(type) {
	case xdr.AccountEntry:
		xdr.NewLedgerEntryData("Account", le)
	case xdr.TrustLineEntry:
		xdr.NewLedgerEntryData("TrustLine", le)
	default:
		panic("unrecognized entry type")
	}
}
```

Comment (leighmcculloch) [reply sdk/state/ingest_test.go]: Assuming it's an update sounds good to me.

Comment (acharb) [reply sdk/state/ingest_test.go]: Since we're requiring it to be a `[]xdr.LedgerEntryData{}` anyways any reason to make the input an interface? The input would be the same. Imo I think it's clearer making the inputs to be a required type 🤔

Comment (leighmcculloch) [reply sdk/state/ingest_test.go]: I was only suggesting so that we don't have to build the ledger entry data manually in tests. In tests I don't think it matters much if we have interface{}'s. Either way, this is great as is.

**Pull request merged**
